### PR TITLE
bulk-downloader-for-reddit: init at 2.6.2

### LINKED ISSUE #314356: Package request: Bulk Downloader for Reddit
**Project description**

Bulk Downloader for Reddit is a tool to download submissions or submission data from Reddit. It can be used to archive data or even crawl Reddit to gather research data. The BDFR is flexible and can be used in scripts if needed through an extensive command-line interface.

**Metadata**

* homepage URL: https://github.com/aliparlakci/bulk-downloader-for-reddit/
* source URL: https://github.com/aliparlakci/bulk-downloader-for-reddit/
* license: GPL v3
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by SkamDart**
## Description of changes

resolves https://github.com/NixOS/nixpkgs/issues/314356

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by SkamDart** [comment]
resolves https://github.com/NixOS/nixpkgs/issues/314356

File: pkgs/by-name/bu/bulk-downloader-for-reddit/package.nix [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      isort
+      pre-commit
+      pytest
+      tox
+    ];
+  };
+
+  pythonImportsCheck = [ "bdfr" ];
+
+  meta = with lib; {
```
Comment (heisfer) [inline pkgs/by-name/bu/bulk-downloader-for-reddit/package.nix:48]: `with lib;` is antipattern. Remove it

**Comment by heisfer** [comment]
> resolves #314356

You need to edit post to take effect https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

**Comment by heisfer** [comment]
Result of `nixpkgs-review pr 315727` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>bulk-downloader-for-reddit</li>
    <li>bulk-downloader-for-reddit.dist</li>
  </ul>
</details>

**Comment by heisfer** [comment]
Weird bug I found while I found while testing it. It wouldn't work unless I added Write permission to
`~/.config/bdfr/default_config.cfg` By default, it only had read permission.
After that I ran `bdfr download ./meh --subreddit nixOS -L 10` and worked just fine.

**Review by heisfer — APPROVED** [review]

**Pull request closed without merging**